### ISSUE TITLE
migrate2rocky fails with "No URLs in mirrorlist"

### ISSUE THREAD

**Issue by meum** [rocky-linux/rocky-tools#153]
```
Preparing to migrate CentOS Linux 8 to Rocky Linux 8.

Determining repository names for CentOS Linux 8.Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

Failed to fetch info for package apr-util-ldap.x86_64.

An error occurred while we were attempting to convert your system to Rocky Linux. Your system may be unstable. Script will now exit to prevent possible damage.

 .Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

Failed to fetch info for package pacemaker-doc.noarch.

An error occurred while we were attempting to convert your system to Rocky Linux. Your system may be unstable. Script will now exit to prevent possible damage.

 .Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

Failed to fetch info for package rootfiles.noarch.

An error occurred while we were attempting to convert your system to Rocky Linux. Your system may be unstable. Script will now exit to prevent possible damage.

 .Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

Failed to fetch info for package epel-release.noarch.

An error occurred while we were attempting to convert your system to Rocky Linux. Your system may be unstable. Script will now exit to prevent possible damage.

 .Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

Failed to fetch info for package libaec-devel.x86_64.

An error occurred while we were attempting to convert your system to Rocky Linux. Your system may be unstable. Script will now exit to prevent possible damage.



Found the following repositories which map from CentOS Linux 8 to Rocky Linux 8:

Getting system package names for CentOS Linux 8Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist

Failed to fetch info for repository .

An error occurred while we were attempting to convert your system to Rocky Linux. Your system may be unstable. Script will now exit to prevent possible damage.
```

**Comment by meum** [comment]
Yes I think so, using vault.centos.org instead seems to work.

```
sed -i /etc/yum.repos.d/CentOS-Linux-*.repo -e 's/mirrorlist=/#mirrorlist=/g' -e 's/#baseurl=http:\/\/mirror/baseurl=http:\/\/vault/g'
```

**Comment by csidirop** [comment]
So replacing with the archive/vault repo let the script do its work?

**Comment by csidirop** [comment]
Thanks, worked without any issues.

**Comment by birkenfeld** [comment]
Would be a nice feature if `migrate2rocky` detected this situation and did the replacements on its own.

**Comment by NeilHanlon** [comment]
We're going to be adding this functionality soon to point to the RL copy of the vault - https://download.rockylinux.org/vault/rocky/

**Comment by pajamian** [comment]
Please use https://dl.rockylinux.org/vault/centos/8.5.2111/* instead of vault.centos.org.  As Neil mentioned we will be updating migrate2rocky to automatically change the CentOS repos to point there in the future.

**Comment by pajamian** [comment]
Fixed with PR #154

**Comment by PaulKGrimes** [comment]
I had to upgrade my Centos * install from 1911 first, to get a recent enough version of dnf. That required me to comment out the mirror lines in the yum.repos.d/* files, and uncomment and edit the baseurl lines to get the Centos updates to work.  After that, the migrate script fails because of that editing.

It'd be great if there was a way for the script to handle migration from somewhat out of date Centos installs.
